src/status/local_settings.nim

```diff
@@ -0,0 +1,78 @@
+import NimQml, os, chronicles
+import status
+import constants as accountConstants
+
+# Local Settings Global keys:
+const LSG_KEY_STORE_TO_KEYCHAIN* = "global/storeToKeychain"
+# Local Settings Global values:
+const LSG_VALUE_STORE* = "store"
+
+const UNKNOWN_ACCOUNT = "unknownAccount"
+
+logScope:
+  topics = "local-settings"
+
+QtObject:
+  type LocalSettings* = ref object of QObject
+    settingsFilePath: string
+    settings: QSettings
+    globalSettingsFilePath: string
+    globalSettings: QSettings
+    
+  proc setup(self: LocalSettings) =
+    self.settingsFilePath = os.joinPath(accountConstants.DATADIR, "qt", UNKNOWN_ACCOUNT)
+    self.settings = newQSettings(self.settingsFilePath, QSettingsFormat.IniFormat)
+    self.globalSettingsFilePath = os.joinPath(accountConstants.DATADIR, "qt", "global")
+    self.globalSettings = newQSettings(self.globalSettingsFilePath, QSettingsFormat.IniFormat)
+    self.QObject.setup
+
+  proc delete*(self: LocalSettings) =
+    self.settings.delete
+    self.globalSettings.delete
+    self.QObject.delete
+
+  proc newLocalSettings*(): LocalSettings =
+    new(result, delete)
+    result.setup
+
+  proc getGlobalSettingsFilePath*(self: LocalSettings): string =
+    return self.globalSettingsFilePath
+
+  proc getSettingsFilePath*(self: LocalSettings): string =
+    return self.settingsFilePath
+
+  proc updateSettingsFilePath*(self: LocalSettings, pubKey: string) =
+    if (pubKey.len > 0):
+      self.settings.delete
+      self.settingsFilePath = os.joinPath(accountConstants.DATADIR, "qt", pubKey)
+      self.settings = newQSettings(self.settingsFilePath, QSettingsFormat.IniFormat)
+
+  proc removeUnknownAccountSettings*(self: LocalSettings) =
```

Comment:
@0kok0 firstly a short explanation: before I added `LocalSettings` class we had only an option to interact with local files from qml, what is not enough for our needs. The plan for the future is to completely remove both `Settings` component (both `globalSettings` and `appSettings`) from the qml side and do store/read/remove properties from local files only via `LocalSettings` class. Till then we're using both approaches.

Once the app is started, before user logged in the app, local file named `unknownAccount` will be created, once used is logged in, that file will be removed and we keep maintaining only two local files `qlobal` and `file_with_user_pk`, that's the reason why we have `removeUnknownAccountSettings` proc.

>  this looks a little dangerous

 What is dangerous?

> Can these functions be called without the user entering the password first?

Why? As I mentioned above `removeUnknownAccountSettings` is called to remove `unknownAccount` local file which we don't use after user is logged in. Also non of local files know for password in any moment. Password is kept in RAM memory for a very short time untill we store it to the Keychain and then it's resetted.

> Also can they be externally modified or only by the (OS) user?

Any local file may be modified by the user, but if he does that maliciously he may just influence to the app settings (like does he want app to be in a dark or light mode, change language and so...), not to any important data like read/change the pass or so.

> Otherwise if the app is unlocked, the user can be locked out of the App or another process can tamper with the settings file s.t. the user potentially looses their funds

How come? This is not possible, try and you will see.

Reply:
Thanks a lot for the elaboration, the purpose of `removeUnknownAccountSettings` to bootstrap the account is now clear to me. 

The scenario I'm referring to here is as follows:
- user chooses store PW in keychain: never, which is persisted 
- setting is changed to "always" without password reprompt and without storing the pw in OS keychain
- app is closed
- app is reopened, but password is not in the keychain
- user needs to manually add a corresponding entry to the OS keychain which can be read by the app, or the app won't open/needs to recovered 

I'd suggest 2 things:
- tighten up the write permissions to the file as much as possible to prevent potential dos/deadlock configs.
- ask for the password (and store it in the keychain) when making changes to "critical" settings (like pw persistence)